src/librustc_middle/query/mod.rs

```diff
@@ -1461,9 +1468,9 @@ rustc_queries! {
             desc { "normalizing `{:?}`", goal }
         }
 
-        query substitute_normalize_and_test_predicates(key: (DefId, SubstsRef<'tcx>)) -> bool {
+        query substitute_impossible_predicates(key: (DefId, SubstsRef<'tcx>)) -> bool {
```

Comment:
Hmm, maybe this would be clearer as `subst_and_check_for_impossible_predicates`?
Or `impossible_predicates_after_subst`?

Reply:
Fixed!